### PR TITLE
Add disable reason to zk storage

### PR DESCRIPTION
**Pull request by EmilyBragg**
Intermediate PR for ZK storing the reason for disable actions.

### CODE REVIEW

File: solrman/smstorage/zk_storage_test.go [hunk 1]
```diff
@@ -120,39 +131,41 @@ func TestZkStorage_IsDisabled(t *testing.T) {
...
-	if s.IsDisabled() {
+	if isDisabled, reason := s.IsDisabled(); isDisabled {
 		t.Error("expected to not be disabled")
 	}
 
-	testutil.create(s.disabledPath())
+	testutil.createWithData(s.disabledPath(), "testDisabled")
 
-	if !s.IsDisabled() {
+	if isDisabled, reason := s.IsDisabled(); !IsDisabled {
```
Comment (thempatel) [inline solrman/smstorage/zk_storage_test.go:140]: we should start to prefer fstesting constructs `fstesting.Equals(t, expected, actual, "msg", args...)`

**Review by thempatel — APPROVED** [review]
LGTM, we might need to run formatting over this, at least on the web view the code looks off

File: solrman/smstorage/zk_storage_test.go (hunk 1 above)
Comment (EmilyBragg) [reply solrman/smstorage/zk_storage_test.go:140]: Yeah. I'm  likely to redo a  chunk of  this with per-org splits/moves etc, so I'll look into refactoring the tests as a part of that :)

Comment (thempatel) [reply solrman/smstorage/zk_storage_test.go:140]: might be blasphemous but should this even be open source? like it has a bunch of stuff specific to us, i dunno how many people are even using this

Comment (EmilyBragg) [reply solrman/smstorage/zk_storage_test.go:140]: I definitely feel like it's way too us-specific to be useful to many folks, and I don't think anyone else uses it really.

**Pull request merged**